code/crates/test/src/validator_set.rs

```diff
@@ -69,6 +69,16 @@ impl ValidatorSet {
         }
     }
 
+    /// Get the number of validators in the set
+    pub fn len(&self) -> usize {
+        self.validators.len()
+    }
+
+    /// Get the validators in the set
+    pub fn get_all(&self) -> Vec<Validator> {
```

Comment:
See comment above. In general let's avoid introducing functions or methods which clone a lot of data under the hood. Not really an issue in the test crate but we should be mindful about this in general :)